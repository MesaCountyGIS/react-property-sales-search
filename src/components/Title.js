import React, { Component } from 'react';

  class Title extends Component {
      constructor(props){
          super(props);

          this.state = {
              isResetButtonVisible: false
          }
      }
    render() {
      return (
          <div className="title" style={{"textAlign": "center", "fontSize":"0.65em"}}>
              <h1>Mesa County Assessor - Property Sales Search</h1>
              <h2>Ken Brownlee, Assessor</h2>
              <div style={{"display":this.state.isResetButtonVisible?"block": "none"}} 
                  id="resetPage">Reset Search</div>
          </div>
      );
    }
  }


export default Title;
