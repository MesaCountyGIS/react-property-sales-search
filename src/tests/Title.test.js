import React from 'react';
import renderer from 'react-test-renderer';
import sinon from 'sinon';
import {stubConsoleError} from './utils/test-utils'
import Title from '../components/Title';

describe("<Title />", ()=>{
    stubConsoleError();

    test('renders without crashing', () => {
        const tree = renderer.create(
            <Title />
      ).toJSON();

      sinon.assert.notCalled(console.error)
      expect(tree).toMatchSnapshot();
    });
});
