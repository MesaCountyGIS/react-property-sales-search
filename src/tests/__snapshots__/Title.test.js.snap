// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Title /> renders without crashing 1`] = `
<div
  className="title"
  style={
    Object {
      "fontSize": "0.65em",
      "textAlign": "center",
    }
  }
>
  <h1>
    Mesa County Assessor - Property Sales Search
  </h1>
  <h2>
    Ken Brownlee, Assessor
  </h2>
  <div
    id="resetPage"
    style={
      Object {
        "display": "none",
      }
    }
  >
    Reset Search
  </div>
</div>
`;
